pymc/distributions/distribution.py

```diff
@@ -343,223 +350,49 @@ def dist(
+class SymbolicRandomVariable(OpFromGraph):
+    """TODO"""
 
-        Returns
-        -------
-        var : TensorVariable
-        """
+    ndim_supp = None
 
-        if "testval" in kwargs:
-            kwargs.pop("testval")
-            warnings.warn(
-                "The `.dist(testval=...)` argument is deprecated and has no effect. "
-                "Initial values for sampling/optimization can be specified with `initval` in a modelcontext. "
-                "For using Aesara's test value features, you must assign the `.tag.test_value` yourself.",
-                FutureWarning,
-                stacklevel=2,
-            )
-        if "initval" in kwargs:
-            raise TypeError(
-                "Unexpected keyword argument `initval`. "
-                "This argument is not available for the `.dist()` API."
-            )
+    def __init__(self, *args, ndim_supp, **kwargs):
+        self.ndim_supp = ndim_supp
+        kwargs.setdefault("inline", True)
+        super().__init__(*args, **kwargs)
 
-        if "dims" in kwargs:
-            raise NotImplementedError("The use of a `.dist(dims=...)` API is not supported.")
-        if shape is not None and size is not None:
-            raise ValueError(
-                f"Passing both `shape` ({shape}) and `size` ({size}) is not supported!"
-            )
-
-        shape = convert_shape(shape)
-        size = convert_size(size)
-
-        create_size, ndim_expected, ndim_batch, ndim_supp = find_size(
-            shape=shape, size=size, ndim_supp=cls.ndim_supp(*dist_params)
-        )
-        # Create the RV with a `size` right away.
-        # This is not necessarily the final result.
-        graph = cls.rv_op(*dist_params, size=create_size, **kwargs)
-
-        # Replicate dimensions may be prepended via a shape with Ellipsis as the last element:
-        if shape is not None and Ellipsis in shape:
-            replicate_shape = cast(StrongShape, shape[:-1])
-            graph = cls.change_size(rv=graph, new_size=replicate_shape, expand=True)
 
-        # TODO: Create new attr error stating that these are not available for DerivedDistribution
-        # rv_out.logp = _make_nice_attr_error("rv.logp(x)", "pm.logp(rv, x)")
-        # rv_out.logcdf = _make_nice_attr_error("rv.logcdf(x)", "pm.logcdf(rv, x)")
-        # rv_out.random = _make_nice_attr_error("rv.random()", "rv.eval()")
-        return graph
+@node_rewriter([SymbolicRandomVariable])
+def inline_SymbolicRandomVariable(fgraph, node):
+    """
+    This optimization expands the internal graph of a SymbolicRV when obtaining logp
+    from Aeppl, in the case it does not have a specialized logp associated with it.
+    """
+    op = node.op
+    # If `Op` is not a `MeasurableVariable`, Aeppl must derive its logp from the inner graph
+    if not isinstance(op, MeasurableVariable):
+        return clone_replace(op.inner_outputs, {u: v for u, v in zip(op.inner_inputs, node.inputs)})
```

Comment:
Two questions here:

1. My understanding is that, if `op` is an instance of a `MeasurableVariable`, it will (and should) have a `logp` dispatched. Will it be of any value of have an `else` statement to check that this is effectively the case in `_logprob.registry`?
2. What does the `clone_replace` here do? Does it emulate what `OpFromGraph` would do, i.e. allowing the log-probability can extract the `logp` of outputs? Can you provide an example of such a case?

Reply:
1. We can
2. This rewrite just does the same as the one in Aesara when OpFromGraph is inlined. This basically includes the inner graph in the original one, so that instead of seeing `CensoredRV`, Aeppl will see the inner `Clip(...)`. That way we can use Aeppl to derive it's logp, even though in PyMC we handle it inside a SymbolicRV
 
https://github.com/aesara-devs/aesara/blob/63f52536a09b13cc55481f071ccb4fc59c9740f4/aesara/compile/builders.py#L932

Reply:
Added a check and test

Reply:
Ended up removing the else branch, otherwise it fails because of the cases where we make the inputs `Unmeasurable`...